lib/java_buildpack/framework/dynatrace_one_agent.rb

```diff
@@ -76,26 +84,39 @@ def supports?
 
       TENANTTOKEN = 'tenanttoken'.freeze
 
-      private_constant :FILTER, :RUXIT_APPLICATION_ID, :RUXIT_CLUSTER_ID, :RUXIT_HOST_ID, :SERVER, :TENANT, :TENANTTOKEN
+      APITOKEN = 'apitoken'.freeze
+
+      ENVIRONMENTID = 'environmentid'.freeze
+
+      ENDPOINT = 'endpoint'.freeze
+
+      private_constant :FILTER, :RUXIT_APPLICATION_ID, :RUXIT_HOST_ID, :SERVER, :TENANT, :TENANTTOKEN, :APITOKEN
+      private_constant :ENVIRONMENTID, :ENDPOINT
 
       def agent_dir
         @droplet.sandbox + 'agent'
       end
 
       def agent_path
-        agent_dir + 'lib64/libruxitagentloader.so'
+        libpath = agent_dir + 'lib64/liboneagentloader.so'
+        libpath = agent_dir + 'lib64/libruxitagentloader.so' unless File.file?(libpath)
+        libpath
       end
 
-      def application_id
-        @application.details['application_name']
+      def agent_download_url
+        credentials = @application.services.find_service(FILTER)['credentials']
+        download_uri = server(credentials).gsub('/communication', '').gsub(':8443', '').gsub(':443', '')
+        download_uri += '/api/v1/deployment/installer/agent/unix/paas/latest?include=java&bitness=64&'
+        download_uri += "Api-Token=#{credentials[APITOKEN]}"
+        ['latest', download_uri]
```

Comment:
Agreed. At the time of compiling the download URL we don't know about the exact version we will get. We only know it is the `latest` version available for the respective environment. We could add a log-message in the `compile` or `release` step tough.